src/main/java/org/spongepowered/common/mixin/tracker/world/end/DragonFightManagerMixin_Tracker.java

```diff
@@ -0,0 +1,39 @@
+package org.spongepowered.common.mixin.tracker.world.end;
+
+import net.minecraft.entity.item.EnderCrystalEntity;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.end.DragonFightManager;
+import net.minecraft.world.end.DragonSpawnState;
+import net.minecraft.world.server.ServerWorld;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.spongepowered.common.event.tracking.PhaseTracker;
+import org.spongepowered.common.event.tracking.context.GeneralizedContext;
+import org.spongepowered.common.event.tracking.phase.world.dragon.DragonPhase;
+
+import java.util.List;
+
+@Mixin(DragonFightManager.class)
+public abstract class DragonFightManagerMixin_Tracker {
+
+    /**
+     * @author i509vcb - February 6th 2020
+     * @reason Add sponge necessary phase state switches
+     *
+     * @param dragonSpawnState The dragon spawn state.
+     * @param worldIn The world this respawnState is occuring in.
+     * @param manager The current DragonFightManager.
+     * @param crystals List of all currently present end crystals.
+     * @param respawnStateTicks The amount of this this respawn state has been running for.
+     * @param exitPortalLocation The position of the exit portal.
+     */
+    @Redirect(method = "tick()V", at = @At(value = "INVOKE", target = "Lnet/minecraft/world/end/DragonSpawnState;process(Lnet/minecraft/world/server/ServerWorld;Lnet/minecraft/world/end/DragonFightManager;Ljava/util/List;ILnet/minecraft/util/math/BlockPos;)V"))
+    private void trackDragonRespawn(DragonSpawnState dragonSpawnState, ServerWorld worldIn, DragonFightManager manager, List<EnderCrystalEntity> crystals, int respawnStateTicks, BlockPos exitPortalLocation) {
```

Comment:
The method name should follow the wiki conventions still: `tracker$wrapSpawnStateWithPhaseEntry` or think of a better name 😆